src/tools/rbd/action/Ubbd.cc

```diff
@@ -0,0 +1,189 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "common/errno.h"
+#include "tools/rbd/Shell.h"
+#include "tools/rbd/Utils.h"
+#include <iostream>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/scope_exit.hpp>
+
+#if defined(WITH_RBD_UBBD)
+extern "C" {
```

Comment:
A good library header should account for the possibility of being included from C++ code.  I would suggest doing this in `libubbd.h` itself.

Reply:
thanx, I will do that in libubbd.h
```
--- a/include/libubbd.h
+++ b/include/libubbd.h
@@ -14,6 +14,11 @@
 
 #ifndef LIBUBBD_H
 #define LIBUBBD_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #include <stdint.h>
 #include <limits.h>
 #include <stdbool.h>
@@ -239,4 +244,9 @@ int ubbd_req_stats(struct ubbd_req_stats_options *opts, struct ubbdd_mgmt_rsp *r
 int ubbd_req_stats_reset(struct ubbd_req_stats_reset_options *opts, struct ubbdd_mgmt_rsp *rsp);
 int ubbd_device_restart(struct ubbd_dev_restart_options *opts, struct ubbdd_mgmt_rsp *rsp);
 int ubbd_device_info(struct ubbd_info_options *opts, struct ubbdd_mgmt_rsp *rsp);
+
+#ifdef __cplusplus
+}
+#endif
+
 #endif /*LIBUBBD_H*/

```